eth/filters/filter.go

```diff
@@ -134,12 +135,12 @@ func (f *Filter) Logs(ctx context.Context) ([]*types.Log, error) {
 		return nil, nil
 	}
 	head := header.Number.Uint64()
-
-	if f.begin == -1 {
+	pending := f.end == rpc.PendingBlockNumber.Int64()
+	if f.begin == rpc.LatestBlockNumber.Int64() || f.begin == rpc.PendingBlockNumber.Int64() {
 		f.begin = int64(head)
 	}
 	end := uint64(f.end)
-	if f.end == -1 {
+	if f.end == rpc.LatestBlockNumber.Int64() || f.end == rpc.PendingBlockNumber.Int64() {
```

Comment:
These need to be fixed, if I request filtering only the pending block, these clauses will force the latest non-pending block to be also included in the dump.